\documentclass[a4paper,11pt,portrait]{article}
\usepackage[MeX]{polski}
\usepackage{polski}
\usepackage{multicol}
\usepackage{listings}
\usepackage{hyperref}
\usepackage{latexsym}
\usepackage{graphicx}
\usepackage{float}
\usepackage{parskip}
\usepackage[utf8]{inputenc}
\usepackage{geometry}
%\geometry{a5paper,top=2cm,bottom=1.5cm,left=-1.3cm,right=0cm}
\title{Package Profile}
\author{Ania Macioszek}
\date{}
%\setlength{\parindent}{50pt}
%\setlength{\parskip}{0pt}

\begin{document}

\maketitle

\section{Introduction}

Package Profile was created to deal with genome profiles,
generated by for example ChIP-seq experiments.
For now, it deals with .wig and .sgr formats.
Package is independent from other BioPython libraries,
so it can be used without the rest of BioPython project.


\section{Usage}

\subsection{Reading from file}

To import Profile library, type

\begin{verbatim}
from Bio import Profile
\end{verbatim}

Now we will be able to create Profile object
by reading some profile from file.
In this tutorial we will use small sample files,
that you will find in [biopython\_directory]/Bio/Profile/.
Currently supported profiles are .wig and .sgr.
To load a file to Profile object, type:

\begin{verbatim}
my_profile = Profile.Profile("biopython/Bio/Profile/sample_profile.sgr")
\end{verbatim}

or

\begin{verbatim}
my_profile = Profile.Profile("biopython/Bio/Profile/sample_profile.wig",
		"wig")
\end{verbatim}

Note that when you load .sgr file you don't need to provide information about used format,
because .sgr is the default one.
Wig files can have some additional data in the first line of the file,
that reading function can load into Profile object.
To suppress this, set readMetadata argument to False.
To provide additional metadata use argument addMetadata.
Metadata should be provided as dictionary.
Example command might look like this:

\begin{verbatim}
my_profile = Profile.Profile("biopython/Bio/Profile/sample_profile.wig,
		"wig", readMetadata = False,
	addMetadata = {"description"="my awesome profile", date="27.07.1410"})
\end{verbatim}

.sgr format doesn't contain any additional data,
 so when format is set to "sgr" argument readMetadata doesn't change behaviour of reading function.
 Of course you can provide additional data anyway.
 If readMetadata is True and metadata is provided with addMetadata argument
 they are both read;
 if some field appears in both of them,
 the one form addMetadata overwrites the one from file.

\subsection{Writing to file}

To write Profile object to file, type

\begin{verbatim}
my_profile.writeProfile("some_profile.sgr")
\end{verbatim}

or

\begin{verbatim}
my_profile.writeProfile("some_profile.wig", "wig")
\end{verbatim}

This will create file some\_profile.sgr (or some\_profile.wig) in the current directory
 in the format specified by the second argument.
 Note that .sgr format is default.


\subsection{Profile object}

Object Profile has following attributes:

	\begin{itemize}

		\item metadata - it's a dictionary storing all metadata about the Profile,
			read from the file or provided later with addMetadata function.
		\item resolution - in case of .wig files,
			resolution is read from the second line
			(it should look like this: "variableStep chrom=chr1 span=50,
			where span is resolution of the profile);
			in case of .sgr files,
			it's estimated from first coordinates.
		\item coordinates - dictionary storing list of coordinates
			for each chromosome,
			for which values are defined
		\item values - dictionary storing list of profile values,
			corresponding to coordinates written in coordinates attribute

	\end{itemize}

addMetadata(dict) function takes as an argument dictionary
 with keys describing names of the fields in metadata
 and values corresponding to their values.
 Note that if key already exists in Profile's metadata
 it will be overwritten.

Sometimes you have couple of profiles you want to analyse,
 all from the same organism,
 yet with different names of chromosomes,
 what can make analysis bit more difficult.
 So, Profile provides function to change chromosome names.
 To see what are the names of the chromosomes in Profile,
 use function getChromosomes().
 To change their names, use function setChromosomeNames(list\_of\_chromosomes).

Sample pipeline with Profile object might look like that:

\begin{verbatim}
>>> from Bio import Profile
>>> my_profile = Profile.Profile("sample_profile.wig","wig")  
>>> my_profile.metadata 
{'name': 'sample profile', 'description': 'some random sample profile'}
>>>my_profile.addMetadata({"replicate" : 1,
	"name"="chip-seq dreb2a binding sites"})
>>> my_profile.metadata
{'name': 'chip-seq dreb2a binding sites',
	'description': 'some random sample profile',
	'replicate': 1}
>>> my_profile.getChromosomes()
['Chr1', 'Chr2', 'Chr3', 'Chr4', 'Chr5', 'Chrc', 'Chrm']
>>> new_chromosomes = ["chrI", "chrII", "chrIII", "chrIV", "chrV", "chrC", "chrM"] 
>>> my_profile.setChromosomeNames(new_chromosomes) 
>>> my_profile.writeProfile("changed_chr_names_and_description.sgr")
\end{verbatim}


\end{document}

